### ISSUE TITLE
Favorites not detected anymore

### ISSUE THREAD

**Issue by june1994** [twintproject/twint#828]
**TLDR**
<img width="325" alt="Capture" src="https://user-images.githubusercontent.com/68325213/87527121-a1e22c80-c6c6-11ea-8d99-80e05465cbfb.PNG">
 - TWINT doesnt seem to work to gather favorites. Other features are not affected. The error message is enclosed.

### Initial Check
- Python version is 3.8
- Installed twint by git clone and pip install requirements
- Doesnt seem to be any duplicates in recent releases.

### Command Ran
twint -u srvmshr --favorites -o output.csv
twint -u srvmshr --favorites >> output.csv
twint -u srvmshr --favorites ---user-full

### Description of Issue
I have been using TWINT since v2.0.0. The favorites could be scraped without problems. I use this to bookmark my articles. Now it doesnt. Other features work i.e. getting the tweets or showing followers.
- I have tried various combinations from the README documents which used to work
- I have tried using last two releases as well. They do not work either.

Error details:
**CRITICAL:root:twint.run:Twint:favorite:favorite_field_lack**
**shit <date field> list index out of range.**

### Environment Details
Using Ubuntu 20.04 natively on two systems which were used to check the systems.

**Comment by yunusemrecatalcam** [comment]
Hi, since twitter changed the ui of the favorites page, old twint was not able to fetch favorites and I have patched in a totally different way(https://github.com/twintproject/twint/pull/826). It seems there is a problem with date parsing, my guess is older tweet's dates include a non standart character (the ? character). Can you share the profile you are trying to fetch favorites?
Also, new patch does not prints favorites to console -I can implement if needed, seems like needed lol-, please try the following code and share if it is working or not.

`import twint

c = twint.Config()
c.Username = "mahfiegilmez"
c.Store_object = True
c.Limit = 20
twint.run.Favorites(c)
out = c.favorited_tweets_list`

**Comment by sourav** [comment]
I tried that too looking at #826, but it seems it doesn't work for me. I tried with the limit set at 200, instead of 20.
PS - The twitter account is mentioned in original issue request. I have tried for  other random accounts as well. Give it a try & let me know if it works?

**Comment by june1994** [comment]
Ok that doesn't work either.

**Comment by yunusemrecatalcam** [comment]
Interesting, it is working correctly on my computer. Now I think it may be caused by a location based behaviour of twitter. For debugging it, you can check the html response on twint/run.py line 45, also you can share the response with me if you want to. https://github.com/twintproject/twint/blob/40c6fd0da88c60069672b70277903afde27744bc/twint/run.py#L45
![Screenshot from 2020-07-19 11-15-43](https://user-images.githubusercontent.com/20454038/87870508-5757fd00-c9b1-11ea-8ed9-9ba8e3c3a674.png)

**Comment by shortdoom** [comment]
@yunusemrecatalcam 

I've checked xpaths and those are obviously fine. I've noticed that ```MobileFav()``` in ```feed.py``` behavior is inconsistent. I've had situations where ```max_id``` was empty, even if it shouldn't. 

```Feed``` and ```favorite``` methods in ```Twint``` class return empty dictionary. 

Testing mobile twitter page with ```bs4``` only using xpaths defined in twint code works without a problem. I think problem lies somewhere in those two methods.

**Comment by itcytandchenmeng** [comment]
> @yunusemrecatalcam
> 
> 我检查了xpaths，那些显然很好。我注意到，`MobileFav()`在`feed.py`行为是不一致的。我曾经遇到过`max_id`空着的情况，即使不应该这样。
> 
> `Feed`而`favorite`在方法`Twint`类返回空字典。（`run.py`）
> 
> `bs4`仅使用twint代码中定义的xpath来测试移动Twitter页面可以正常工作。我认为问题出在这两种方法中。

I want to know how you corrected it?
